oop/src/PurchaseOrders/use-cases/SubmitPO.usecase.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { createUuid } from "../../utilities/uuid";
+import { PONumber } from "../domain/PONumber.value-object";
+import { PurchaseOrder } from "../domain/PurchaseOrder.entity";
+import { PORepository } from "../infrastructure/repositories/PurchaseOrder.repo";
+import { SubmitPO } from "./SubmitPO.usecase";
+
+describe("SubmitPO", () => {
```

Comment:
I created a separate `SubmitPO` use case. This use case is currently only responsible for incrementing the po number. But based on some hints I assume we'll need a submit flow anyway. 

> I want Purchase Orders to be automatically assigned a Purchase Order number when they're submitted

I made a choice not to have the `CreatePO` use case accept a po number at this point. So in order for one to be assigned the purchase order needs to be submitted. This means a PO Number _should_ always be unique because it's up to this use case to assign it (and it's incremented here). I'm unsure if that will cause problems down the road. I can see scenarios where the `CreatePO` use case would take a full purchase order. But I figure I can refactor at that point if it's an issue 🤔